pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/TextIndexCreatorProvider.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.creator;
+
+import java.io.IOException;
+import org.apache.pinot.segment.spi.index.creator.TextIndexCreator;
+
+
+public interface TextIndexCreatorProvider {
+  /**
+   * Creates a {@see TextIndexCreator} from information about index creation.
+   * This allows a plugin to pattern match index creation information to select
+   * an appropriate implementation.
+   * @param context context about the index creation.
+   * @return a {@see ForwardIndexCreator}
+   * @throws IOException whenever something goes wrong matching or constructing the creator
+   */
+  TextIndexCreator newTextIndexCreator(IndexCreationContext context)
```

Comment:
All current three impls (Lucene text, lucene fst, native fst) implement TextIndexCreator interface. So, can we try to keep a single `newTextIndexCreator` and let the `IndexCreationContext` dictate which impl to create ?

Reply:
Not really, not without adding more information to `IndexCreatorContext` because of the way it has been purposefully kept separate, which has rippled into the configuration layer. What I think would make sense is not to allow overriding of the Lucene based text index at all (it's also awkward to do for the reader), and leave it completely untouched by this change. At some point in the future we can make a decision about whether and how to migrate the first incarnation of text index to the FST indexes, but leave this distinction out of this SPI for now.